src/github.com/matrix-org/dendrite/common/config/config.go

```diff
@@ -444,6 +412,7 @@ func (e Error) Error() string {
 
 // check returns an error type containing all errors found within the config
 // file.
+// nolint: gocyclo
```

Comment:
please don't do this. Factor out a separate function to keep the cyclomatic complexity count down.

Reply:
This function will likely keep growing with the configuration.
Rather than just putting the verification of the logging config in an external function, what about having one function per first-order element in the config ?
Something like `checkMatrix` which checks parameters `matrix.*`, `checkMedia` which checks parameters `media.*`, and so on.
Essentially, refactoring the entire `check` function rather than just moving the last loop.